src/sophys/common/plans/parallel_wrapper.py

```diff
@@ -30,9 +32,10 @@ def parallel_plans_wrapper(*args):
                     if current_thread[list_id] == None:
                         # Yield one message or yield it in a thread in case of a 'wait' message
                         current_event[list_id] = next(parallel_plan_list[list_id])
-                        if current_event[list_id][0] == 'wait':
+                        if current_event[list_id][0] == 'wait' or current_event[list_id][0] == 'sleep':
                             current_thread[list_id] = Thread(
-                                target=yield_msg, args=(current_event[list_id]), daemon=True)
+                                target=asyncio.run, args=(yield_msg(current_event[list_id]), ))
```

Comment:
Instead of launching an asyncio context every time there's a message, could we do something using [`asyncio.Runner`](https://docs.python.org/3/library/asyncio-runner.html#asyncio.Runner)? That way you can submit the wait/sleep functions to the same runner.